### DISCUSSION TITLE
Foreach fails to write browser cookies in batches, the first cookie in the dictionary will be written successfully, and the subsequent ones will fail to write anyway;

### DISCUSSION THREAD

**Discussion by yy520a** [cefsharp/CefSharp#3696] [General]
//CookieList这里通常会加载10到15个cookie键值对，而且没有包含任何非法字符；
               public static Dictionary<string, string> CookieList = new Dictionary<string, string>();
               ...........................................................
               //这里清理全局cookie是为了方便测试所添加的，可能与问题无关；
               Cef.GetGlobalCookieManager().DeleteCookiesAsync(null, null); 

                foreach (KeyValuePair<string, string> item in CookieList)
                {
                    var cookie = new CefSharp.Cookie
                    {
                        Domain = ".baidu.com",
                        Name = item.Key,
                        Value = item.Value,
                        Expires = DateTime.Now.AddDays(720)
                    };
                    Cef.GetGlobalCookieManager().SetCookie("https://www.baidu.com", cookie);
                }

               **If the Name and Value fields in this foreach are changed to absolute value types, a single write is no problem, but this loses its meaning. If it is taken from the dictionary or referenced variables will fail to write, but first A cookie key value will be written successfully, and subsequent writes will fail;**

               这段foreach 里的Name与Value字段如果改成绝对值类型，单次写入是没问题的，但是这样便失去了意义，如果是从字典里取出或者引用的变量都会写入失败，但第一个cookie键值会写入成功，后续的都会写入失败；

**Comment by amaitland** [comment]
You'll need to discuss this on https://magpcss.org/ceforum/index.php

CefSharp is just one of many chromium embedded framework wrappers.

**Reply by amaitland** [reply]
Also English only please (same for ceforum).

**Reply by yy520a** [reply]
Okay thank you!

**Reply by amaitland** [reply]
One thing to try would be to await the delete cookies method call and use http://cefsharp.github.io/api/91.1.x/html/M_CefSharp_AsyncExtensions_SetCookieAsync.htm awaiting each set cookie call. You aren't waiting for each cookie to be written before trying to write the next one.

There is no concept of batching to my knowledge. 

If that doesn't work then asking on ceforum is the way to go.
